skeleton/app.js

```diff
@@ -121,19 +121,23 @@ export default class App {
     applySingleInstance() {
         if ('singleInstance' in this.settings && this.settings.singleInstance) {
             this.l.verbose('setting single instance mode');
-            const isSecondInstance = app.makeSingleInstance(() => {
-                // Someone tried to run a second instance, we should focus our window.
-                if (this.window) {
-                    if (this.window.isMinimized()) {
-                        this.window.restore();
-                    }
-                    this.window.focus();
-                }
-            });
 
-            if (isSecondInstance) {
+            const isFirstInstance = app.requestSingleInstanceLock();
+
+            if (!isFirstInstance) {
                 this.l.warn('current instance was terminated because another instance is running');
                 app.quit();
+            } else {
+                app.on('second-instance', () => {
+                    // Someone tried to run a second instance, we should focus our window.
+                    if (this.window) {
+                        if (this.window.isMinimized())
+                            this.window.restore();
+
+                        this.window.show();
+                        this.window.focus();
```

Comment:
why do we need three of them here?
show contains focus action, I guess

https://www.electronjs.org/docs/api/browser-window#winshow

```suggestion
                        if (this.window.isMinimized())
                            this.window.restore();
                            this.window.focus();
                        else
                            this.window.show();
```